integrations/db/h2/pom.xml

```diff
@@ -46,6 +46,7 @@
         <dependency>
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
+            <scope>provided</scope>
```

Comment:
A question: my understanding is this Maven module exists so that users who want to use H2 in native image can just grab it and forget about it. My guess is that's why H2 is/was in `compile` scope here. Should it really be `provided` now?